source/fe/fe_values.cc

```diff
@@ -510,8 +510,8 @@ namespace FEValuesViews
                                                shape_derivatives[0].size() : derivatives.size();
       AssertDimension (derivatives.size(), n_quadrature_points);
 
-      std::fill (derivatives.begin(), derivatives.end(),
-                 typename ProductType<Number,dealii::Tensor<order,spacedim> >::type());
+      typedef typename ProductType<Number,dealii::Tensor<order,spacedim> >::type ProductTensorType;
```

Comment:
why not just use `typename derivatives::value_type`?

Reply:
True, that would definitely be more concise. If we choose to follow through with this PR then I'll change this first.